File: VB/CustomizeTransition.Module.Win/Controllers/TransitionCustomizationController.vb
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Win.SystemModule
Imports DevExpress.Utils.Animation

Namespace CustomizeTransition.Module.Win
Public Class TransitionCustomizationController
    Inherits WindowController

    Private officeNavBarTransitionController As OfficeNavBarTransitionController

    Private Sub OnCustomizeTransition(ByVal sender As Object, ByVal e As CustomizeTransitionEventArgs)
        e.TransitionManager.Transitions(e.TransitionControl).TransitionType = New CombTransition()
    End Sub
    Protected Overrides Sub OnActivated()
        MyBase.OnActivated()
        Dim officeNavBarTransitionController As OfficeNavBarTransitionController = Frame.GetController(Of OfficeNavBarTransitionController)()
        If officeNavBarTransitionController IsNot Nothing Then
            AddHandler officeNavBarTransitionController.CustomizeTransition, AddressOf OnCustomizeTransition
        End If
    End Sub
    Protected Overrides Sub OnDeactivated()
        If officeNavBarTransitionController IsNot Nothing Then
            RemoveHandler officeNavBarTransitionController.CustomizeTransition, AddressOf OnCustomizeTransition
        End If
        MyBase.OnDeactivated()
    End Sub
    Public Sub New()
        TargetWindowType = WindowType.Main
    End Sub
End Class
End Namespace

File: CS/CustomizeTransition.Module.Win/Controllers/TransitionCustomizationController.cs
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Win.SystemModule;
using DevExpress.Utils.Animation;

namespace CustomizeTransition.Module.Win
{
public class TransitionCustomizationController : WindowController
{
    private OfficeNavBarTransitionController officeNavBarTransitionController;

    private void OnCustomizeTransition(object sender, CustomizeTransitionEventArgs e) {
        e.TransitionManager.Transitions[e.TransitionControl].TransitionType = new CombTransition();
    }
    protected override void OnActivated() {
        base.OnActivated();
        OfficeNavBarTransitionController officeNavBarTransitionController = Frame.GetController<OfficeNavBarTransitionController>();
        if(officeNavBarTransitionController != null) {
            officeNavBarTransitionController.CustomizeTransition += OnCustomizeTransition;
        }
    }
    protected override void OnDeactivated() {
        if(officeNavBarTransitionController != null) {
            officeNavBarTransitionController.CustomizeTransition -= OnCustomizeTransition;
        }
        base.OnDeactivated();
    }
    public TransitionCustomizationController() {
        TargetWindowType = WindowType.Main;
    }
}
}
